src/app/utils/storage.js

```diff
@@ -0,0 +1,30 @@
+export const STORAGE_KEY = "bookmarkData"
+
+function getData() {
+  if (typeof window === "undefined") return {}
+  const raw = localStorage.getItem(STORAGE_KEY)
+  return raw ? JSON.parse(raw) : { bookmarks: {} }
+}
+
+function setData(data) {
+  localStorage.setItem(STORAGE_KEY, JSON.stringify(data))
```

Comment:
Handle QuotaExceededError as well. Something like this
```suggestion
 try {
  localStorage.setItem(key, JSON.stringyfy(data));
} catch (e) {
  if (e.name === 'QuotaExceededError') {
    console.error('Storage is full! Please clear some data.');
    // TODO: Inform the user or implement a strategy to free up space
  } else {
    // Some other error (e.g., disabled localStorage)
    console.error('Error writing to storage:', e);
  }
}
```

Reply:
Hi Triet, I would like to notify both of you that "please don't do this". The `storage.js` file is a service, and it SHOULD NOT be responsible for handling exceptions. Therefore, don't handle anything here unless you have a very good reason to. The better place to handle the exception is in the page.

Reply:
That make sense. I'll resolve this comment and create a separate task for that. Thanks, Nathan.